libs/dal/src/repositories/topic/topic-subscribers.schema.ts

```diff
@@ -31,7 +31,7 @@ const topicSubscribersSchema = new Schema(
       index: true,
       required: true,
     },
-    subscribers: [{ type: Schema.Types.ObjectId, ref: 'Subscriber', required: true }],
+    subscribers: [{ type: Schema.Types.String, field: 'subscriberId', ref: 'Subscriber', required: true }],
```

Comment:
Might have missed it from previous PR's, but why not doing an object mapping with a _subcriberId on each topic subscriber entity instead of having all subscribers in the array here? I'm afraid we might be limited by mongo object size for very large topics.

So basically meaning to follow a more SQL-ish pattern for topic subscribers collection. So each subscriber will have his own document for the subscription. It might be better because that we will be able to save metadata for the topic subscription for individual subscriber, for example specific user preferences or other metadata. 

Or am I missing something here?

Reply:
Ok. Let's change the structure of the document.